draft-ietf-taps-impl.md

```diff
@@ -410,6 +388,30 @@ When ordering branches, an implementation can give more weight to properties tha
 The available protocols and paths on a specific system and in a specific context can change; therefore, the result of sorting and the outcome of racing may vary, even when using the same Selection and Connection Properties. However, an implementation ought to provide a consistent outcome to applications, e.g., by preferring protocols and paths that are already used by existing Connections that specified similar Properties.
 
 
+## Candidate Gathering {#gathering}
+
+The step of gathering candidates involves identifying which paths, protocols, and endpoints may be used for a given Connection. This list is determined by the requirements, prohibitions, and preferences of the application as specified in the Selection Properties.
+
+### Gathering Endpoint Candidates
+
+Both Local and Remote Endpoint Candidates must be discovered during connection establishment.  To support Interactive Connectivity Establishment (ICE) {{?RFC8445}}, or similar protocols that involve out-of-band indirect signalling to exchange candidates with the Remote Endpoint, it’s important to be able to query the set of candidate Local Endpoints, and give the protocol stack a set of candidate Remote Endpoints, before it attempts to establish connections.
```

Comment:
/it's/it is/
is /be able to/ needed?
/give the protocol stack/provide the protocol stack with/
/it attempts/the Local Endpoint attempts/

Reply:
"be able to" is actually doing something here but I won't quibble. made all these in a suggested change.